python/cudf/cudf/core/buffer/spillable_buffer.py

```diff
@@ -27,6 +37,84 @@
 T = TypeVar("T", bound="SpillableBuffer")
 
 
+def get_spillable_owner(data) -> Optional[SpillableBuffer]:
+    """Get the spillable owner of `data`, if any exist
+
+    Search through the stack of data owners in order to find an
+    owner of type `SpillableBuffer` (not subclasses).
+
+    Parameters
+    ----------
+    data : buffer-like or array-like
+        A buffer-like or array-like object that represent C-contiguous memory.
+
+    Return
+    ------
+    SpillableBuffer or None
+        The owner of `data` if spillable or None.
+    """
+
+    if type(data) is SpillableBuffer:
+        return data
+    if hasattr(data, "owner"):
+        return get_spillable_owner(data.owner)
+    return None
+
+
+def as_spillable_buffer(data, exposed: bool) -> SpillableBuffer:
+    """Factory function to wrap `data` in a SpillableBuffer object.
+
+    If `data` isn't a buffer already, a new buffer that points to the memory of
+    `data` is created. If `data` represents host memory, it is copied to a new
+    `rmm.DeviceBuffer` device allocation. Otherwise, the memory of `data` is
+    **not** copied, instead the new buffer keeps a reference to `data` in order
+    to retain its lifetime.
+
+    If `data` is owned by a spillable buffer, a "slice" of the buffer is
+    returned. In this case, the spillable buffer must either be "exposed" or
+    spilled locked (called within an acquire_spill_lock context).
```

Comment:
Can we add a note clarifying why this is necessary? Is it because otherwise, the owner could be spilled during the execution of this function?

Reply:
Added 
```
    If `data` is owned by a spillable buffer, a "slice" of the buffer is
    returned. In this case, the spillable buffer must either be "exposed" or
    spilled locked (called within an acquire_spill_lock context). This is to
    guarantee that the memory of `data` isn't spilled before this function gets
    to calculate the offset of the new slice.
```